snap_plugin/v1/plugin.py

```diff
@@ -363,8 +388,20 @@ def start_plugin(self):
             sys.stdout.flush()
 
     def _generate_preamble_and_serve(self):
-        self._port = self.server.add_insecure_port('127.0.0.1:{!s}'.format(0))
-        self.server.start()
+        if self._args.tls_enabled:
+            try:
+                self._tls_setup()
+                credentials = self._generate_tls_credentials()
+                self._port = self.server.add_secure_port('127.0.0.1:{}'.format(self._port), credentials)
+                LOG.info("Configured secure port on {}.".format(self._port))
+                self.server.start()
+            except Exception as e:
+                LOG.error(str(e))
+                LOG.error("TLS setup failed. Unable to add secure port.")
```

Comment:
We should disable possibility to run HTTP server here. Currently even if configuration fails server is up and running.
current behavior:
```
python examples/collector/rand.py --stand-alone --stand-alone-port 8282 --tls
2017-08-03 09:24:22,412 - snap_plugin.v1.plugin - ERROR - private-key argument is missing. Required with tls-enabled.
2017-08-03 09:24:22,412 - snap_plugin.v1.plugin - ERROR - TLS setup failed. Unable to add secure port.
Plugin loaded at 0.0.0.0:8282
```
expected behavior:
```
python examples/collector/rand.py --stand-alone --stand-alone-port 8282 --tls
2017-08-03 09:25:07,374 - snap_plugin.v1.plugin - ERROR - private-key argument is missing. Required with tls-enabled.
2017-08-03 09:25:07,374 - snap_plugin.v1.plugin - ERROR - TLS setup failed. Unable to add secure port.
2017-08-03 09:25:07,374 - snap_plugin.v1.plugin - ERROR - Unable to run stand-alone.
```